NBitcoin/BitcoinCore/Coins.cs

```diff
@@ -88,58 +63,43 @@ public Coins(Transaction tx, int height)
 
 		private void UpdateValue()
 		{
-			_Value = vout.Where(o => !IsNull(o))
+			Value = Outputs.Where(o => !IsNull(o))
 							.Select(o => o.Value).Sum();
 		}
 
-		private bool IsNull(TxOut o)
-		{
-			return o.Value.Satoshi == -1;
-		}
-
-		public bool IsEmpty
-		{
-			get
-			{
-				return vout.Count == 0;
-			}
-		}
+		private bool IsNull(TxOut o) => o.Value.Satoshi == -1;

```

Comment:
Guys, i see two issues here. 


1. I think  that it is TxOut responsibility to tell wheter it is Null. 
2. **IsNull** seems to technical term here.  What business name would be more suitable here? **HasSatoshies** ? Other?

So instead of 
```
private void UpdateValue()
		{
			Value = Outputs
				.Where(o => !IsNull(o))
				.Sum(o=> o.Value);
		}
```

I suggest 

```
private void UpdateValue()
		{
			Value = Outputs
				.Where(o => !o.IsNull())
				.Sum(o=> o.Value);
		}
```

Or even better without negation
```
private void UpdateValue()
		{
			Value = Outputs
				.Where(o => o.HasSatoshies())
				.Sum(o=> o.Value);
		}
```